### PR TITLE
VCB-215: make compare deep equal wherever possible

### PR DESCRIPTION
**Pull request by deflox**

### CODE REVIEW

File: src/board/alu.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export function sub(w1: Word, w2: Word): AluResult {
   flags[Flag.C] =
     w1.toUnsignedInteger() + comp.toUnsignedInteger() > Word.MAX_VALUE
   flags[Flag.V] =
-    w1.hasSign() != w2.hasSign() && w1.hasSign() != result.hasSign()
+    w1.hasSign() != w2.hasSign() && w1.hasSign() !== result.hasSign()
```
Comment (PhilSchni) [inline src/board/alu.ts:59]: Why only with result.hasSign() and not with w.2hasSign() as well?

Comment (deflox) [reply src/board/alu.ts:59]: yes you are right I missed that one. Will change that.

**Review by m-dot-wigger — CHANGES_REQUESTED** [review]
there are also 2 unreplaced == 
(in components>memory>index.tsx and
instruction>opcode.ts)

File: src/board/alu.ts (hunk 1 above)
Comment (deflox) [reply src/board/alu.ts]: Changed

**Comment by deflox** [comment]
> there are also 2 unreplaced == (in components>memory>index.tsx and instruction>opcode.ts)

Changed those places too.

**Review by m-dot-wigger — APPROVED** [review]
now all looks good to me

**Pull request merged**